plugins/modules/postgresql_pg_hba.py

```diff
@@ -747,6 +925,103 @@ def user_weight(self):
         return 1
 
 
+def _strip_quotes(string):
+    if not string:
+        return string
+    return string[1:-1] if string.startswith("\"") else string
+
+
+def parse_auth_options(options):
+    option_dict = {}
+    for option in options:
+        split_option = OPTION_RE.match(_strip_quotes(option))
+        if not split_option:
+            raise PgHbaRuleValueError(
+                "Found invalid option '{}'. Options need to be in the format 'key=value'".format(option))
+        if split_option.group(1) in option_dict.keys():
+            raise PgHbaRuleValueError(
+                "The rule contains two options with the same key ('{0}')".format(split_option.group(1)))
+        option_dict[split_option.group(1)] = split_option.group(2)
+
+    return option_dict
+
+
+def handle_db_and_user_strings(string):
+    # if the string is quoted or a regex, we return it unaltered
+    if "\"" in string or string.startswith("/"):
+        return string
+    # we sort the dbs/users alphabetically
+    else:
+        return ",".join(sorted(string.split(",")))
+
+
+def handle_address_field(address):
```

Comment:
Can this part be simplified by letting [`ipaddress.ip_network`](https://docs.python.org/3/library/ipaddress.html#ipaddress.ip_network) do the parsing?

Reply:
Not really, I could try to parse it with `ipaddress` and then catch the exception if it doesn't work, but I'm not a big fan of "control-flow by exception", so I avoid that when possible.  
Also, everything that is not an IP, may be a hostname, but that would for example not be allowed to contain colons. And last, but not least, I wanted to provide more specific error-messages than "that's invalid". So I need some finer checks, in any case.

Reply:
I'm asking because you want to add >60 lines of code for what I think could be done in much less. I prefer less code because it's easier to understand and maintain.
I also dislike "control-flow by exception" but in this case I see no harm.
How about this?
```python3
if FQDN_REGEX.match(address):
  return address, "hostname", -1
try:
  address = ipaddress.ip_network(address)
except ValueError as e:
  raise PgHbaRuleValueError("Adress '{0}' is not a valid network: {1}".format(address, e))
return str(address.network_address), "IPv" + str(address.version), address.prefixlen
```
The most complex part of this proposal would be the `FQDN_REGEX` (see https://stackoverflow.com/questions/11809631/fully-qualified-domain-name-validation).

Reply:
I changed it, it is a little shorter now